### PR TITLE
Application tier redress

### LINKED ISSUE #521: Terraform: Update Naming conventions for Application Tier resources
# Problem statement

The Application Tier resources added in #515 assumed a pattern for the resource names. This pattern should be updated to follow a convention to clearly indicate relevant information for the resource.

## Enhancement

Update the resource names to align with the Naming Standards document.

The decision has been made for top level resources (i.e. resources which appear in the portal) are to be named in the format `SID_(scs|app)##[-subType]`

### LINKED ISSUE #566: Topics for Redress from 20200616
TODO's
```
Application Tier
	Rename files
		App-vm.tf --> vm-app.tf
		Scs-vm.tf --> vm-scs.tf

	Move resources
		nsg/nsr --> nsg-app-subnet.tf
		ALB --> infrastructure.tf
		SCS avset --> infrastructure.tf
		APP avset --> infrastructure.tf

	Change resource
		VM OS version bound to latest; Needs to be parameterized to allow for version pinning.

	Missing Resources
		SCS Disk (500G) & Attachment
		APP Disk (500G) & Attachment

	Naming
		Object ID's
			Recursion - the resource type does not need to be part of the object name. Simplify; Consistency. (ex: in vm-scs.tf all objectID's should be scs) (ex: resource "azurerm_network_interface" "scs" { )

		Resource Names
			Standardize
				Prepend SID to Resource Name
				Count in name should be formatted; (ex: ${format("%02d", count.index)} )
				Add comment above any name assignment to reflect naming standard used.
ex:
# Naming Standard: {SID}_app##-subnet
name = "${upper(var.application.sid)}-app${format("%02d", count.index)}-nic"

				Consider naming standards and how there nest to always reflect a easily identifiable and unique representation. Below is an example thought process. We are not there yet, just food for thought.
				# Naming Standard: {DEPLOYZONE}-{REGION_MAP}-SAP{LOGICAL_NETWORK}-{PROJECT_NAME}_{LANDSCAPE}-{SID}_app-subnet

			NIC
				ip_configuration block
					name - make name static;  Unique naming not required; Descriptive of purpose maybe, not necessary.
					examples:
						IPCONFIG1, FRONTEND, BACKEND, MANAGEMENT
				
		
```

### PR DESCRIPTION
**Pull request by peterchoo**
# Problem

The work to add the Application tier received some overall review notes from @mkdeegan in #566.

These notes cover work as described in #517 and #521.

## Description

As all the items in #566 are effected by the file rename, they have all been included in one PR.

Closes #566, closes #521, initial work towards #517.

## Review Notes

1. `scs-vm.tf` and `app-vm.tf`  renamed to `vm-scs.tf` and `vm-app.tf` to group them together.
1. `resource azurerm_*` object IDs changes to `scs` and `app` to remove repetition.
1. Azure Load Balancer and Availability Sets have been moved to the `infrastructure.tf` file.
1. NSG definition has been moved to a new `nsg-app-subnet.tf` file.
1. Azure Resource names updated to format `SID_(scs|app)##[-subType]` where the resources will appear in the Azure Portla.
1. SCS/App NICs IP configuration given a static name of `IPConfig1`
1. SCS/App VMs have a 512GB data disk added, with default configuration driven from a local variable for easier updating in the future.
1. Application Tier OS image specification now has an optional `version` parameter which defaults to `latest`.

### CODE REVIEW

File: deploy/terraform/modules/app_tier/variables_local.tf [hunk 1]
```diff
@@ -80,4 +82,12 @@ locals {
     62000 + tonumber(local.scs_instance_number),
     62100 + tonumber(local.ers_instance_number)
   ]
+
+  # Define options for Data Disks
+  data-disk = {
+    size_gb           = 512
+    disk_type         = "Premium_LRS"
+    caching           = "None"
+    write_accelerator = false
```
Comment (rukawata) [inline deploy/terraform/modules/app_tier/variables_local.tf:91]: I think with issue #517, we would have a json to define the sku, disk specs as well as amount of data disk required, just like hdb_size.json.
It is okay for this PR, maybe as backlog item for next week?

File: deploy/terraform/modules/app_tier/vm-scs.tf [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    name                 = "${upper(var.application.sid)}_scs${format("%02d", count.index)}-osDisk"
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  source_image_reference {
+    publisher = local.os.publisher
+    offer     = local.os.offer
+    sku       = local.os.sku
+    version   = "latest"
```
Comment (rukawata) [inline deploy/terraform/modules/app_tier/vm-scs.tf:52]:
```suggestion
    version   = local.os.version
```

File: deploy/terraform/modules/app_tier/nsg-app-subnet.tf [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Creates SAP app subnet nsg
+resource "azurerm_network_security_group" "nsg-app" {
+  count               = local.enable_deployment ? (var.infrastructure.vnets.sap.subnet_app.nsg.is_existing ? 0 : 1) : 0
+  name                = var.infrastructure.vnets.sap.subnet_app.nsg.name
+  location            = var.infrastructure.region
+  resource_group_name = var.vnet-sap[0].resource_group_name
```
Comment (rukawata) [inline deploy/terraform/modules/app_tier/nsg-app-subnet.tf:6]: Maybe I missed this in the previous review.
NSG should be deployed in the rg that is defined in the input but not tight to the vnet :)
```suggestion
  location                      = var.resource-group[0].location
  resource_group_name           = var.resource-group[0].name
```

File: deploy/terraform/modules/app_tier/vm-scs.tf (hunk 2 above)
Comment (peterchoo) [reply deploy/terraform/modules/app_tier/vm-scs.tf]: Good catch, another victim of the rebase. I'll fix the App file as well.

File: deploy/terraform/modules/app_tier/nsg-app-subnet.tf (hunk 3 above)
Comment (peterchoo) [reply deploy/terraform/modules/app_tier/nsg-app-subnet.tf]: Good catch, I just moved the code to the appropriate file.

**Pull request merged**